src/modules/accommodations/components/accommodation-card/accommodation-card.tsx

```diff
@@ -57,12 +45,60 @@ const styles = {
   ratingStar: { fontSize: '17px' },
 };
 
-export function AccommodationCard({ status, id, pricePerNight, address, name, rating, image }: AccommodationCardProps) {
-  const [addToWishlist, setAddToWishlist] = useState(true);
+export function AccommodationCard({
+  status,
+  id,
+  pricePerNight,
+  address,
+  name,
+  rating,
+  image,
+  isSavedToWishlist,
+}: AccommodationCardProps) {
+  const [isWishlisted, setIsWishlisted] = useState(isSavedToWishlist);
+  const { wishlistMutation } = useWishlistMutation();
+  const authGuardAction = useAuthGuardAction();
+
+  function handleWishlistAction() {
+    if (!id) {
+      return;
+    }
+
+    setIsWishlisted((prev) => !prev);
+
+    wishlistMutation.mutate(
+      {
+        data: {
+          accommodationId: id,
+        },
+        action: isWishlisted ? 'remove' : 'add',
+      },
+      {
+        onSuccess: (data) => {
+          const message = data ? 'Accommodation was added to wishlist' : 'Accommodation was removed from wishlist';
+          enqueueSnackbar(message, {
```

Comment:
Create these properties as a shared config. Or better create a custom function that returns configured enqueueSnackbar.

Reply:
> Create these properties as a shared config. Or better create a custom function (optionally separate for each alert type: success, info, error) that returns configured enqueueSnackbar.

Done